sql/snapshot.sql

```diff
@@ -0,0 +1,100 @@
+-- public.global_snapshot definition
+
+-- Drop table
+
+-- DROP TABLE global_snapshot;
+
+CREATE TABLE global_snapshot (
+	ordinal int8 NOT NULL,
+	hash text NOT NULL,
+	height int8 NOT NULL,
+	subheight int8 NOT NULL,
+	last_snapshot_hash text NULL,
+	created_at timestamp NULL,
+	updated_at timestamp NULL,
+	CONSTRAINT global_snapshot_pkey PRIMARY KEY (ordinal)
+);
+CREATE INDEX global_snapshot_hash_idx ON public.global_snapshot USING btree (hash);
+
+
+-- public.dag_balance_change definition
+
+-- Drop table
+
+-- DROP TABLE dag_balance_change;
+
+CREATE TABLE dag_balance_change (
+	ordinal int8 NOT NULL,
+	address text NOT NULL,
+	balance int8 NULL,
+	CONSTRAINT dag_balance_change_pk PRIMARY KEY (ordinal, address),
+	CONSTRAINT dag_balance_change_global_snapshot_fk FOREIGN KEY (ordinal) REFERENCES global_snapshot(ordinal)
+);
+
+
+-- public.dag_transaction definition
+
+-- Drop table
+
+-- DROP TABLE dag_transaction;
+
+CREATE TABLE dag_transaction (
```

Comment:
Notes: 
- `hash` can be VARCHAR
- `global_ordinal` - nice, let’s use this format throughout. Need an index on this column to fetch transactions per snapshot.
- `block` - is this supposed to be `block_hash`? Can be VARCHAR.
- `source` - this is a wallet address. Should be VARCHAR.
- `destination` - also a wallet address
- `block_hash` - what’s the difference between this and `block`?
- `transaction_original_id` - does this FK to a separate table? I’m not really sure why we return this from the API but we’ll need to continue returning it for backwards compatibility. We could store it as JSON here as it’s never queried separately from this record and it’s a nested record so it would mean multiple tables to store it separately.
- `created_at` / `updated_at` - same as above

Reply:
There are a few other missing columns here if you check against the existing BE API output: salt, parent, etc.

https://be-mainnet.constellationnetwork.io/transactions?limit=10